src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -110,12 +110,14 @@ const predicate_map = {
 
 const condition_fit_functions = {
     's3:ExistingObjectTag': _is_object_tag_fit,
-    's3:x-amz-server-side-encryption': _is_server_side_encryption_fit
+    's3:x-amz-server-side-encryption': _is_server_side_encryption_fit,
+    's3:VersionId': _is_object_version_fit
 };
 
 const supported_actions = {
     's3:ExistingObjectTag': ['s3:DeleteObjectTagging', 's3:DeleteObjectVersionTagging', 's3:GetObject', 's3:GetObjectAcl', 's3:GetObjectTagging', 's3:GetObjectVersion', 's3:GetObjectVersionTagging', 's3:PutObjectAcl', 's3:PutObjectTagging', 's3:PutObjectVersionTagging'],
-    's3:x-amz-server-side-encryption': ['s3:PutObject']
+    's3:x-amz-server-side-encryption': ['s3:PutObject'],
+    's3:VersionId': ['s3:GetObjectVersion', 's3:DeleteObjectVersion', 's3:GetObject']
```

Comment:
this condition is not supported for GetObject. But it is supported for GetObjectVersionAttributes, GetObjectVersionTagging, PutObjectVersionTagging, DeleteObjectVersionTagging

Reply:
done